### PR TITLE
Rewrite problematic language where possible

### PR DESCRIPTION
**Pull request by mrunge**
This patch leaves out rewrite master/slave from mysql and also
master from snmp.

This repaces blacklist with blocklist, whitelist with allowlist.

ChangeLog: collectd: Fix problematic use of language where possible.

### CODE REVIEW

File: docs/BUILD.dpdkstat.md [hunk 1]
```diff
@@ -217,7 +217,7 @@ See also: http://dpdk.org/doc/guides/prog_guide/multi_proc_support.html
 
  *  The same PCI device configuration should be passed to the primary process as
     the secondary process uses the same port indexes as the primary.
- *  A blacklist / whitelist of NICs isn't supported yet.
+ *  A deny-list / allow-list of NICs isn't supported yet.
```
Comment (igorpeshansky) [inline docs/BUILD.dpdkstat.md:220]: I believe the prevalent terminology is `s/blacklist/blocklist/`, and `s/whitelist/allowlist/` (no dashes in "allowlist") — see, e.g., [this article](https://www.theregister.com/2019/09/03/chromium_microsoft_offensive/).

Comment (mrunge) [reply docs/BUILD.dpdkstat.md:220]: Thank you for the quick review, I'll update the PR

**Review by igorpeshansky — DISMISSED** [review]
LGTM :shipit:

**Comment by mrunge** [comment]
For master/slave (for mysql), would everybody agree with using primary/replica.

This is a change included in mariadb https://mariadb.com/kb/en/standard-replication, I've added that to this change here as well.

File: src/mysql.c [hunk 2]
```diff
@@ -463,40 +463,40 @@ static int mysql_read_slave_stats(mysql_database_t *db, MYSQL *con) {
...
     mysql_free_result(res);
     return -1;
   }
 
-  if (db->slave_stats) {
+  if (db->replica_stats) {
     unsigned long long counter;
     double gauge;
 
-    gauge_submit("bool", "slave-sql-running",
```
Comment (octo) [inline src/mysql.c:477]: Changing the metric names breaks backwards compatibility. Can we ear-mark removing this language from the metric identity for version 6 please?

**Review by octo — CHANGES_REQUESTED** [review]

**Comment by octo** [comment]
Thank you so much for working on this Matthias! Unfortunately changing the metric identity is problematic, as it breaks backwards compatibility. In the upcoming version 6 release we should be able to replace those terms in the metric identities, too.

File: src/mysql.c [hunk 3]
```diff
@@ -193,11 +193,11 @@ static int mysql_config_database(oconfig_item_t *ci) /* {{{ */
     else if (strcasecmp("ConnectTimeout", child->key) == 0)
       status = cf_util_get_int(child, &db->timeout);
     else if (strcasecmp("MasterStats", child->key) == 0)
-      status = cf_util_get_boolean(child, &db->master_stats);
+      status = cf_util_get_boolean(child, &db->primary_stats);
```
Comment (mrunge) [inline src/mysql.c:196]: I shouldn't change this here, since it's still used in mysql/mariadb in older releases.

File: src/mysql.c (hunk 2 above)
Comment (mrunge) [reply src/mysql.c]: you are right, this is fixed in the next commit.

**Review by octo — APPROVED** [review]
Thanks @mrunge!

**Pull request merged**